Interface fixes

zerofixgiven opened the pull request:
## Motivation or cause

Interface has some non uniform parts and clipping also some things were interfering with cp times window in TM2.

## Change description

Adjusted UI elements

## Checklist of pull request

Make sure that your pull request follow the following 'rules':

- I have read the **CONTRIBUTING** document.
- My code follows the code style of this project.
- All new and existing tests passed, except in few situations.
- My change requires a change to the documentation.

zerofixgiven commented:
I don't know why appveyor fails on view.py I did only change the value 109.5 to 110.5

tomvlk commented:
Hi @zerofixgiven, can you update your pull request with the current master branch? I already fixed some in the meantime.

zerofixgiven commented:
@tomvlk Should be done, man I am so bad at using git 🤣 

I updated it to my latest versions, which also added:
- format fixes in some files
- option to disable karma announce
- fixes in dedimania and local_records to handle if the nickname of a player ends with a single $

tomvlk commented:
Good work, I will test this and then merge when all okay, maybe with some corrections.
Thank you for your work!

zerofixgiven commented:
@tomvlk Ok, thank you very much!

I've also prepared a change for a menu (also funcmd) with 2 rows to avoid overlap with list views.

But currently if one opens both menus (main and funcmd) at the same time it will overlap, I don't know much about manialink programming, so therefore if someone has a clue how to trigger a close event or something on the other menu (main >< funcmd) would be appreciated! :)

w1lla commented:
Hi,

To make it so that only PlayerToolBar is active or Funcmd use the following code:

https://gist.github.com/w1lla/3841e3e85625d264600ded3a0029239a toolbar.Script.txt
https://gist.github.com/w1lla/5a12cc71c40ba9a57316d318e7028712 emoji_toolbar.Script.txt

This will only let the player open 1 toolbar, so if funcmd is used, people can't use the player toolbar and vice-versa.

zerofixgiven commented:
@w1lla Thank you so much! I'll will test my prepared change with your scripts then! 🙂

zerofixgiven commented:
I've made some more adjustments:

- ui position fixes
- convert player & funcmd menu to two rows to avoid overlapping with the LiveViews (see screenshot below)
- add ability to ensure that only 1 menu is open by force hiding the other menu (Thanks to w1lla for providing information about sharing variables between manialink scripts.)
- 
![grafik](https://user-images.githubusercontent.com/56362033/102690940-1aff2600-4209-11eb-951a-d7b4b919dc6f.png)

TheMaximum commented on pyplanet/apps/contrib/dedimania/__init__.py:
```diff
@@ -586,7 +586,10 @@ async def chat_current_record(self):
 			message = '$0b3Current Dedimania Record: $fff\uf017 {}$z$s$0b3 by $fff{}$z$s$0b3 ($l[http://dedimania.net/tm2stats/?do=stat&UId={}&Show=RECORDS]$fff{}$0b3 records$l)'.format(
-				times.format_time(first_record.score), first_record.nickname, self.instance.map_manager.current_map.uid, records_amount
+				times.format_time(first_record.score),
+				(lambda x: x if not (x[-1] == '$' and x[-2:].count('$') == 1) else x[:-1])(first_record.nickname),
```
I suspect this is to avoid styling to continue into the rest of the text?
Is there anything specific that the $z$s does not catch?

zerofixgiven replied:
This is correct.
I noticed that there will be an issue if the name of the player ends with a single $.
In this case $$z will be interpreted as $z which leads to not resetting the formating.

For dedimania the file __init__.py is probably the only location to catch this.

TheMaximum replied:
Ah, I wasn't aware of that! Thanks for your response 👍

TheMaximum commented on pyplanet/apps/contrib/local_records/__init__.py:
```diff
@@ -299,7 +299,9 @@ async def chat_current_record(self):
 			message = '$0f3Current Local Record: $fff\uf017 {}$z$s$0f3 by $fff{}$z$s$0f3 ($fff{}$0f3 records)'.format(
-				times.format_time(first_record.score), first_record.player.nickname, records_amount
+				times.format_time(first_record.score),
+				(lambda x: x if not (x[-1] == '$' and x[-2:].count('$') == 1) else x[:-1])(first_record.player.nickname),
```
(Same question as before.)

zerofixgiven replied:
(Same answer as before.)

@tomvlk 

For local_records __init__.py might not be the best location - I am not sure why I did put it there, but since the player name will be derived from the database, it might be better to save the nickname to the database without a leading $ if there is one. Then there has to be a update player check with adjusted nickname too.

TheMaximum reviewed (commented):
Thanks for your changes! Could you provide a full screenshot of all the changes you've made?

zerofixgiven commented:
Screenshot of PyPlanet 0.9.5 without UI modifications (both menus opened):

![image](https://user-images.githubusercontent.com/56362033/108630914-a98d5f00-7467-11eb-8b1d-b6976a0b86f6.png)

Screenshot of PyPlanet 0.9.5 with UI modifications (main menu opened):

![image](https://user-images.githubusercontent.com/56362033/108630827-29ff9000-7467-11eb-97ae-de89f7e36348.png)

Screenshot of PyPlanet 0.9.5 with UI modifications (funcmd menu opened):

![image](https://user-images.githubusercontent.com/56362033/108630863-574c3e00-7467-11eb-8065-50de39114dd2.png)

tomvlk commented:
@zerofixgiven How do you feel about updating it once more, we will seriously check it out afterwards? We had to implement small bugfixes and small features instead in the meantime. I think we can finally focus on fine tuning on the interface. Sorry for the delays.

zerofixgiven commented:
@tomvlk No worries, I can understand that interface is not the top priority. I am willing to update it, but I can not specify a time frame.

tomvlk commented:
@zerofixgiven no problem mate, I'm already happy that you are willing to spend time again into this topic.

zerofixgiven commented:
@tomvlk Hehe, my pleasure. Unfortunately I suck at git - I guess that will be the biggest hurdle. :')

zerofixgiven commented:
@tomvlk I hope I did it right.

Couldn't test it though. I had an issue with a migration of the jukebox addon.
![grafik](https://user-images.githubusercontent.com/56362033/200131630-0c9fb540-e8e7-444a-b68f-4ebff5f6b35d.png)

After disabling jukebox another error with `RuntimeWarning: coroutine 'Pool.from_pool_fill' was never awaited`.
![grafik](https://user-images.githubusercontent.com/56362033/200131684-88d47edf-9156-429c-9a8c-941d2fac665b.png)

Did not have time to dig into the issues. Do you know what causes the issue?
I did try Python 3.7.9 and Python 3.9.7.

tomvlk commented:
@TheMaximum could you check what gone wrong in the migration possibly?
@zerofixgiven the first error is due to a migration, I asked @TheMaximum to check a little bit with this. The second issue, I highly suggest you to change to MySQL (the first issue might be related as well).
We kind of dropped PostgreSQL support as it is barely used, and difficult to support with native queries.

zerofixgiven commented:
@tomvlk That is quite unfortunate. All my servers use PostgreSQL.

zerofixgiven commented:
@tomvlk disabled jukebox and applied #1126. With that I was able to start it and it does seem to work. Did not really click through everything though.

I noticed that the PyPlanet and Discord logos do overlap and that the old Discord logo is used.

Did just have a quick look over the code.
Is dropping PostgreSQL due to native queries mainly because of the new app "rankings"?

tomvlk commented:
@zerofixgiven yes this is the main reason. Also the ranking apps. Native queries are so much faster in many areas that we couldn't do without it. 
How many servers are you running on PostgreSQL? This migration is to be honest a mistake and we forgot to write the query in both languages. If you can and know how to you could write the query here that we can use instead and I'll write a fix for it.

tomvlk commented:
@zerofixgiven could you try if it works with this change:

https://github.com/PyPlanet/PyPlanet/pull/1230

zerofixgiven commented:
@tomvlk I checked it out, after that I got another error, which I could fix. I commented in the Code tab of #1230.

tomvlk commented:
@zerofixgiven new version is out with a bugfix for this :-)

zerofixgiven commented:
@tomvlk I created a new environment installed 0.10.4 and applied the changes. After starting I noticed that aiopg version 1.2.1 was installed. I thought that was fixed with #1189, but noticed that It was not merged in 0.10.4. Manually downgraded to 0.16.0 changes do work. Looks pretty exactly like what you can see in the pictures from 21 Feb 2021.

I also updated the admin toolbar because my OCD kicked in. Here a gif for comparison:
![toolbar comparison](https://user-images.githubusercontent.com/56362033/200377495-7740cddc-5845-458a-9527-464d745c36cd.gif)

As mentioned above:
> I noticed that the PyPlanet and Discord logos do overlap and that the old Discord logo is used.

![grafik](https://user-images.githubusercontent.com/56362033/200351716-6d89bd9d-e888-4ad7-b8b1-5c842a310fbe.png)

Since I can not change the logo I would guess that changing the logo based on the [Discord Branding](https://discord.com/branding) and adding padding between the logos would be something you would need to do, right?

tomvlk commented:
Hi @zerofixgiven 
I'm about to merge but found out that the label is not correct on the admin toolbar for the two replay/restart icon:
![image](https://github.com/PyPlanet/PyPlanet/assets/3877688/0b0081f1-15cb-4eff-a699-aa6bb995614c)

zerofixgiven commented:
@tomvlk Thank you for finding this, I've rebased the branch and fixed the copy & paste error.

Verified the fix before rebasing. But could not start pyplanet after the rebase anymore, though.
![grafik](https://github.com/PyPlanet/PyPlanet/assets/56362033/162c0053-9fcb-4db6-93f5-d2824639bea9)

Did not work with a fresh installation of pyplanet in python 3.7.9 either therefore I did not try to chase the rabbit knowing the issue is fixed.

I am running a postgres database on all my servers - mentioning this because this caused issues last time too.
